### PR TITLE
SONARJAVA-4160 Refactor S5868 using sonar-analyzer-commons

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**
The implementation is in fact almost the same, it's just that the helper class code has been moved around.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,7 +82,7 @@
 
     <sonar.version>9.4.0.54424</sonar.version>
     <sonarlint.plugin.api.version>6.3.0.36253</sonarlint.plugin.api.version>
-    <analyzer.commons.version>1.24.0.969</analyzer.commons.version>
```
Comment (quentin-jaquier-sonarsource) [inline pom.xml:85]: TODO: update this version to a one from master...

Comment (m-g-sonar) [reply pom.xml:85]: PR on commons merged!

File: java-checks/src/main/java/org/sonar/java/checks/regex/CanonEqFlagInRegexCheck.java [hunk 2]
```diff
@@ -93,7 +94,9 @@ private static class CharacterVisitor extends RegexBaseVisitor {
 
     @Override
     public void visitSequence(SequenceTree tree) {
-      subjectToNormalization.addAll(getGraphemeInList(tree.getItems()));
+      subjectToNormalization.addAll(getGraphemeInList(tree.getItems()).stream()
```
Comment (m-g-sonar) [inline java-checks/src/main/java/org/sonar/java/checks/regex/CanonEqFlagInRegexCheck.java:97]: what about not collecting into a list but directly add in `subjectToNormalization`?
```java
getGraphemeInList(tree.getItems())
  .stream()
  .map(RegexIssueLocation::fromCommonsRegexIssueLocation)
  .forEach(subjectToNormalization::add)
```

**Review by m-g-sonar — COMMENTED** [review]
LGTM, except that we can probably slightly simplify the collection of the grapheme.

File: java-checks/src/main/java/org/sonar/java/checks/regex/CanonEqFlagInRegexCheck.java (hunk 2 above)
Comment (quentin-jaquier-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/regex/CanonEqFlagInRegexCheck.java]: Indeed, way better!

File: pom.xml (hunk 1 above)
Comment (quentin-jaquier-sonarsource) [reply pom.xml:85]: I updated the version to rely on `1.24.0.973`.

**Pull request merged**